packages/scenario/src/client/answerOf.ts

```diff
@@ -0,0 +1,29 @@
+import { UnreachableError } from '@odk-web-forms/common/lib/error/UnreachableError.ts';
+import type { AnyNode, RootNode, AnyLeafNode as ValueNode } from '@odk-web-forms/xforms-engine';
+import { SelectNodeAnswer } from '../answer/SelectNodeAnswer.ts';
+import { StringNodeAnswer } from '../answer/StringNodeAnswer.ts';
+import type { ValueNodeAnswer } from '../answer/ValueNodeAnswer.ts';
+import { getNodeForReference } from './traversal.ts';
+
+const isValueNode = (node: AnyNode): node is ValueNode => {
+	return node.nodeType === 'select' || node.nodeType === 'string';
+};
+
+export const answerOf = (instanceRoot: RootNode, reference: string): ValueNodeAnswer => {
```

Comment:
❓ does this return the value when the referenced node is not relevent?

Reply:
It returns a `ValueNodeAnswer` for any leaf/value node with the specified `reference`, regardless of its relevance state. If I recall, this originally only looked up nodes produced by [`getPositionalEvents`](https://github.com/getodk/web-forms/blob/d40bd88ed8959bbe7fae5d28efc840c16ca50a72/packages/scenario/src/jr/event/getPositionalEvents.ts#L46) (or what eventually became that). It had to be expanded to search all nodes for:

- Many calls to look up/assert the "answer of" model-only nodes, which are not produced as `PositionalEvent`s because they're not "questions". We may want to reconsider the _name_ of this function on that basis alone (but we'd want to consider how that affects portability with JavaRosa, whether JR should rename too). I see 40 tests affected by this.

- In one case (that I can find now, could've sworn there were more!), JavaRosa calls `answerOf` for a [surprising assertion](https://github.com/getodk/web-forms/blob/d40bd88ed8959bbe7fae5d28efc840c16ca50a72/packages/scenario/test/relevant.test.ts#L563-L565) of the non-blank(!) value of an explicitly non-relevant node. From what I can tell, this means that JavaRosa treats non-relevance of values as an "engine concern" (in our parlance) for XPath expressions, but evidently as a "client concern" (our parlance) in terms of the actual presence of a value otherwise.

Based on the notes taken at the time, it seemed like we had consensus that the surprising assertion is concerned with implementation details, and not one we should treat as gospel. That being the case though, my instinct is to keep that test parameterized, and assert the node is present and its client-facing value **is blank** within our interface.

I think that would be a valuable assertion of the current behavior anyway. I also think that there's a potential future where we may actually want to produce both behaviors—where certain clients (e.g. internal debug tooling) may want to inspect and surface the retained value of a node that's currently treated as blank due to non-relevance.